dtsync/cbor_gen.go

```diff
@@ -18,15 +18,22 @@ var _ = cid.Undef
 var _ = math.E
 var _ = sort.Sort
 
-var lengthBufMessage = []byte{131}
-
 func (t *Message) MarshalCBOR(w io.Writer) error {
 	if t == nil {
 		_, err := w.Write(cbg.CborNull)
 		return err
 	}
-	if _, err := w.Write(lengthBufMessage); err != nil {
-		return err
+	if t.OrigPeer == "" {
+		var lengthBufMessage = []byte{131}
+		if _, err := w.Write(lengthBufMessage); err != nil {
+			return err
+		}
+	} else {
+		var lengthBufMessage = []byte{132}
+
+		if _, err := w.Write(lengthBufMessage); err != nil {
+			return err
+		}
```

Comment:
These are manual edits right? 

Is there any chance we can move these out of `_gen` file into a separate go file in the same package so that `go generate` can't override them?

If we can't then I recommend remocing `_gen` from the name and heavily document the rationale behind modifying generated code.

Reply:
I implemented the two message structs + wrapper, and afterward, decided that more code actually made it less clear to me. It also required copying the data to deserialize since deserialization consumes the buffer.

I opted to use non-generated code as this was less code to maintain, clearer (with comments added), and slightly more efficient.  Also, the message should at some point pre replaced with an IPLD schema and derialization done using bindnode.